### DISCUSSION TITLE
Creation of sub-sub-section fails

### DISCUSSION THREAD

**Discussion by meepmeep** [oetiker/SmokePing#330] [Q&A]
I currently have a quite simple configuration with multiple section containing each few monitor.

```
*** Targets ***

probe = FPing

menu = Top
title = Network Latency Grapher
remark = Smokeping Firefly ! \
         Here you will learn all about the latency of our network.

slaves = kif

##
## Server
##

+ Server
menu = Server
title = Server
alerts = someloss,hostdown
probe = FPing

++ Jarvis
title = Jarvis
host = xx.xxx.xx
menu = Jarvis

++ GLaDOS
title = GLaDOS
host = xx.xxx.xx
menu = GLaDOS

++ Firefly
title = Firefly
host = xx.xxx.xx
menu = Firefly

++  MultiHosts
menu = MultiHosts
title = Multihosts Server
host = /Server/Jarvis /Server/GLaDOS /Server/Firefly

++ MultiHostsKif
menu = MultiHosts Kif
title = Multihosts Server from Kif
host = /Server/Firefly~kif /Server/Morty~kif /Server/jcd~kif /Server/plouf~kif

++ MultiHostsVPS
menu = MultiHosts VPS
title = Multihosts VPS
host = /Server/kif /Server/jcd /Server/plouf

++ kif
title = Kif-FR
host = xx.xxx.xx
menu = Kif-FR
slaves =

++ jcd
title = jcd-NL
host = xx.xxx.xx
menu = jcd-NL

++ plouf
title = plouf-CH
host = xx.xxx.xx
menu = plouf-CH
```

As soon as I tried to add a new "VPS" sub section under Server, it crashed :  

```
Software error:
Can't call method "_pings" on an undefined value at /usr/share/perl5/Smokeping.pm line 889.
For help, please send mail to this site's webmaster, giving this error message and the time and date of the error. 
```

Config file : 

```
(previous same as above with MultiHostsVPS etc.)

++ VPS
title = VPS
menu = VPS
alerts = someloss,hostdown
probe = FPing

+++ kif
title = Kif-FR
host = xx.xxx.xx
menu = Kif-FR
slaves =

+++ jcd
title = jcd-NL
host = xx.xxx.xx
menu = jcd-NL

+++ plouf
title = plouf-CH
host = xx.xxx.xx
menu = plouf-CH
```

**Comment by cnighswonger** [comment]
```
++ VPS
title = VPS
menu = VPS
alerts = someloss,hostdown
probe = FPing
```
This appears to be missing a 'host' keyword.

What version of smokeping are you running?

Also, why do you duplicate the probe definition in this subsection? It is defined as FPing at the top level of your Targets configuration.

**Reply by meepmeep** [reply]
The host line is not mandatory in a section. 
Moreover VPS is just the "head name" of the section. There is not host link to it. 

For the probe, it was just an attempt to fix this (and it failed)

Version is the lastest debian stable package (2.7.3-3 - package version)

**Reply by cnighswonger** [reply]
> The host line is not mandatory in a section. Moreover VPS is just the "head name" of the section. There is not host link to it.

My bad. I read `+++` rather than `++`

When you added the VPS section, did you modify the pathing in the multihost list for this target?

```
++ MultiHostsVPS
menu = MultiHosts VPS
title = Multihosts VPS
host = /Server/kif /Server/jcd /Server/plouf
```

**Reply by meepmeep** [reply]
Yes, I thought of that too. But commenting this part didn't do the job either.

**Reply by meepmeep** [reply]
I created a fully new fake section + sub section, and it worked perfectly.
I think that smokeping doesn't like when I move existing hosts from one section to another (maybe because of duplicates node name ?).

**Reply by cnighswonger** [reply]
I have noticed that smokeping does not seem to clean up after itself when one moves/renames various nodes.
